src/viking.js

```diff
@@ -1,11 +1,138 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health;
+        this.strength = strength;
+    }
 
+    attack() {
+        return this.strength;
+    }
+
+    receiveDamage(damage) {
+        this.health = this.health - damage;
+    }
+
+}
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength){
+        super(name, health, strength);
+        this.name= name;
+        this.health = health;
+        this.strength = strength;
+    }
+
+    receiveDamage(damage) {
+        this.health = this.health - damage;
+            if (this.health <= 0) {
+                return `${this.name} has died in act of combat`; 
+            } else {
+        return `${this.name} has received ${damage} points of damage`;
+        }
+    }
+
+    battleCry() {
+        return "Odin Owns You All!";
+    }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    constructor(health, strength){
+        super(health, strength);
+        this.health= health;
+        this.strength= strength;
+    }
+
+    receiveDamage(damage) {
+        this.health = this.health - damage;
+        if (this.health <= 0) {
+            return "A Saxon has died in combat"; 
+        } else {
+        return `A Saxon has received ${damage} points of damage`;
+        }
+    }
+}
 
 // War
-class War {}
+class War {
+    constructor(){
+        this.vikingArmy=[];
+        this.saxonArmy=[];
+    }
+
+    addViking(Viking) {
+        // this.vikingArmy +=;
+    }
+
+    addSaxon(Saxon) {
+
+    }
+
+    vikingAttack() {
+        let attackedSaxon = Math.floor(Math.random(this.saxonArmy));
+        receiveDamage(damage) 
+    }
+
```

Comment:
It looks like you started building the function but didn't know where to go. Here's an example of how the vikingAttack can be made. Let me know if you have questions about it
```
vikingAttack() {
    let randomVikingIndex = Math.floor(Math.random() * this.vikingArmy.length);
    let randomViking = this.vikingArmy[randomVikingIndex];
    let randomSaxonIndex = Math.floor(Math.random() * this.saxonArmy.length);
    let randomSaxon = this.saxonArmy[randomSaxonIndex];
    let thingToReturnAtTheEnd = randomSaxon.receiveDamage(
      randomViking.attack()
    );
    if (randomSaxon.health <= 0) {
      this.saxonArmy.splice(randomSaxonIndex, 1);
    }

    return thingToReturnAtTheEnd;
  }